### PR TITLE
fix(flutter): various grid ui issues

### PR DESCRIPTION
**Pull request by richardshiue**
1. fix this monstrosity

    Screenshot 2024-09-04 at 11 48 39 AM

    Screenshot 2024-09-04 at 5 18 26 PM

3. On desktop, make enter on checklist items end editing, only allow shift + enter to go to a new line
4. don't animate the wrap text toggle when opening it
5. add integration test for when changing field type during creation
6. allow opening a related database from the popover

    https://github.com/user-attachments/assets/31e94760-de5d-40a4-a2c0-0db0838e9c40

### Feature Preview

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

**Comment by LucasXu0** [comment]
@richardshiue How about adding a 'before' and 'after' comparison?

We have a confirmation dialog style. You may replace the existing one in the database with it.

`showConfirmDeletionDialog`, `showConfirmDialog` and `showCancelAndConfirmDialog`.

File: frontend/appflowy_flutter/lib/plugins/database/application/field/field_editor_bloc.dart [hunk 1]
```diff
@@ -175,7 +184,7 @@ class FieldEditorEvent with _$FieldEditorEvent {
 @freezed
 class FieldEditorState with _$FieldEditorState {
   const factory FieldEditorState({
-    required final FieldInfo field,
+    required final FieldInfo? field,
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database/application/field/field_editor_bloc.dart:187]:
```suggestion
    @Default(null) FieldInfo? field,
```

**Review by Xazin — APPROVED** [review]

**Comment by annieappflowy** [comment]
Number 2: shift + enter -> new line
This is inconsistent with other blocks across mobile and desktop.
Since Mobile is difficult to have the equivalence of shift + enter, we have changed the logic of adding a new line for both desktop and mobile apps.
- Enter -> New line within the same block (quote, callout, etc)
- Shift + Enter / Tap on mobile -> Exit the current block and go to the next line

File: frontend/appflowy_flutter/lib/plugins/database/application/field/field_editor_bloc.dart (hunk 1 above)
Comment (richardshiue) [reply frontend/appflowy_flutter/lib/plugins/database/application/field/field_editor_bloc.dart:192]: Decided to not feed null into constructor, should I still do this regardless?

Comment (Xazin) [reply frontend/appflowy_flutter/lib/plugins/database/application/field/field_editor_bloc.dart:192]: It's nullable, so we might as well default it I guess.

Comment (richardshiue) [reply frontend/appflowy_flutter/lib/plugins/database/application/field/field_editor_bloc.dart]: Sure thing

**Pull request merged**